frontend/src/pages/search/index.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import { GetServerSideProps } from 'next'
+import Link from 'next/link'
+import { SWRConfig } from 'swr'
+import { unstable_serialize } from 'swr/infinite'
+
+import Page from '@components/Page'
+import Project from '@models/Project'
+import {
+  makeSWRKeyGetter,
+  searchFetcher,
+  useLazySearch,
+} from '@hooks/useLazySearch'
+import ProjectCardGrid from '@components/ProjectCardGrid'
+import { useSearchQuery } from '@contexts/SearchContext'
+
+import styles from './index.module.scss'
+
+interface SearchPageProps {
+  initialQuery: string
+  swrFallback: Record<string, Array<Array<Project>>>
+}
+
+export const getServerSideProps: GetServerSideProps = async ({
+  query: queryParams,
+}) => {
+  const query = (queryParams.q as string) || ''
+
+  const searchParams = { query }
+
+  const hits = await searchFetcher(searchParams)
+
+  return {
+    props: {
+      swrFallback: {
+        [unstable_serialize(makeSWRKeyGetter(searchParams))]: [hits],
+      },
+      initialQuery: query,
+    } as SearchPageProps,
```

Comment:
Doesn't this type assertion circumvent type checking?

Reply:
Converted it to normal annotation in cedd4fd, the assigned object gets type checked.